src/api/db/models/Image.ts

```diff
@@ -143,16 +143,45 @@ export class ImageModel {
     context: Pick<Context, 'user'>,
   ): Promise<gql.StandardErrorPayload> {
     try {
-      const res = await Promise.allSettled(
-        input.imageIds!.map((imageId) => {
-          return this.deleteImage({ imageId }, context);
+      const s3 = new S3.S3Client({ region: process.env.AWS_DEFAULT_REGION });
+      console.time('delete-images total');
+      console.time('delete-images mongo records');
+      const images = await Image.find({ _id: { $in: input.imageIds! } });
+
+      const res = await Promise.allSettled([
+        await Image.deleteMany({
+          _id: { $in: input.imageIds! },
+          projectId: context.user['curr_project'],
+        }),
+        await ImageAttempt.deleteMany({
+          _id: { $in: input.imageIds! },
+          projectId: context.user['curr_project'],
+        }),
+        await ImageError.deleteMany({
+          image: { $in: input.imageIds! },
+        }),
+      ]);
+      console.timeEnd('delete-images mongo records');
+
+      const keys: { Key: string }[] = [];
+      console.time('delete-images s3 records');
+      images!.forEach((image) => {
```

Comment:
So kinda stuck on this bit. Should we only delete images that we can find mongo records for or should we just delete all images that were inputted by the user?